flutter_src/lib/logic/state/bloc_validator.dart

```diff
@@ -1,20 +1,37 @@
-import 'package:macrobaseapp/logic/usecases/macro_firestore/firestore_macro_operation.dart';
+import 'package:macrobaseapp/logic/api/firestore_db.dart';
 import 'package:macrobaseapp/model/entities/macro.dart';
 
 class CustomBlocValidator {
   static Future<String> nameValidator(String name) async {
+    FirestoreService db = FirestoreService();
+
     //Avoid too many Firestore calls
     await Future.delayed(Duration(milliseconds: 500));
-    List<Macro> list = await queryMacro(name);
+    List<Macro> list = await db.queryMacro(name);
     if (list.length > 0) {
       return "Sorry, this macro name already exist";
```

Comment:
Sorry for missing this before, but overall this is not a good validator structure. The validator should only return true/false. This way it can be reused in other tasks. Defining the specific string that is displayed is the job of the caller. This applies to all the validators here.

Reply:
Agree to simplify validator output to boolean. Consider adding a logger to log details.